File: inventario/src/main/java/com/joyeria/inventario/service/UsuarioService.java
package com.joyeria.inventario.service;

import com.joyeria.inventario.entity.Usuario;
import java.util.List;
import java.util.Optional;

public interface UsuarioService {
    Usuario crear(Usuario usuario);
    List<Usuario> listarTodos();
    Optional<Usuario> buscarPorId(String id);
    Optional<Usuario> buscarPorEmail(String email);
    Usuario actualizar(String id, Usuario usuario);
    void eliminar(String id);
}

File: joyeria-frontend/src/app/core/services/usuario.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class UsuarioService {
private apiUrl = 'https://inventario-backend-0lj9.onrender.com/api/usuarios';

  constructor(private http: HttpClient) {}

  getPerfil(): Observable<any> {
    return this.http.get(`${this.apiUrl}/perfil`);
  }

  actualizarPerfil(datos: any): Observable<any> {
    return this.http.put(`${this.apiUrl}/perfil`, datos);
  }

  cambiarPassword(datos: any): Observable<any> {
    return this.http.put(`${this.apiUrl}/cambiar-password`, datos);
  }

  listarTodos(): Observable<any[]> {
    return this.http.get<any[]>(this.apiUrl);
  }

  crear(usuario: any): Observable<any> {
    return this.http.post(`${this.apiUrl}/registro`, usuario);
  }

  actualizar(id: string, usuario: any): Observable<any> {
    return this.http.put(`${this.apiUrl}/${id}`, usuario);
  }

  eliminar(id: string): Observable<any> {
    return this.http.delete(`${this.apiUrl}/${id}`);
  }
}